aspen/simplates/__init__.py

```diff
@@ -11,99 +11,149 @@
+        get the response to a request for this page
+
+        accept - an HTTP Accept: header asking for this page
+        state - a dict of execution context values you wish to supply
+                * Note that these are overriden by values that are carried
+                over from the execution of the zeroth page
+        """
+
+        # copy the state dict to avoid accidentally mutating it
+        spt_context = dict(state)
+        # override it with values from the first page
+        spt_context.update(self.pages[0])
+        # use this as the context to execute the second page in
+        exec(self.pages[1], spt_context)
 
         if '__all__' in spt_context:
             # templates will only see variables named in __all__
             spt_context = dict([ (k, spt_context[k]) for k in spt_context['__all__'] ])
 
-        return self.get_response(state, spt_context)
+        # negotiate or punt
+        render, media_type = self.pages[2]  # default to first content page
+        if accept is not None:
+            try:
+                media_type = mimeparse.best_match(self.available_types, accept)
+            except:
+                # exception means don't override the defaults
+                log( "Problem with mimeparse.best_match(%r, %r): %r "
+                   % (self.available_types, accept, sys.exc_info())
+                    )
+            else:
+                if media_type == '':    # breakdown in negotiations
+                    raise SimplateException(self.available_types)
+                render = self.renderers[media_type] # KeyError is a bug
+
+        body = render(spt_context)
+        return media_type, body
 
 
-    def parse_into_pages(self, raw, is_bound):
-        """Given a bytestring and a boolean, return a list of pages.
+    def parse_into_pages(self, raw):
+        """Given a bytestring that is the entire simplate, return a list of pages.
         """
 
         pages = list(split_and_escape(raw))
         npages = len(pages)
 
         # Check for too few pages.
-        if npages < self.min_pages:
+        if npages < MIN_PAGES:
             type_name = self.__class__.__name__[:-len('resource')]
             msg = "%s resources must have at least %s pages; %s has %s."
             msg %= ( type_name
-                   , ORDINALS[self.min_pages]
+                   , _ordinal(MIN_PAGES)
                    , self.fs
-                   , ORDINALS[npages]
+                   , _ordinal(npages)
                     )
             raise SyntaxError(msg)
 
         # Check for too many pages. This is user error.
-        if self.max_pages is not None and npages > self.max_pages:
+        if MAX_PAGES is not None and npages > MAX_PAGES:
             type_name = self.__class__.__name__[:-len('resource')]
             msg = "%s resources must have at most %s pages; %s has %s."
             msg %= ( type_name
-                   , ORDINALS[self.max_pages]
+                   , _ordinal[MAX_PAGES]
```

Comment:
ah, good catch.  Not caught because that code is never run because MAX_PAGES is currently hardcoded to None.